### PR TITLE
Added support for iptables physdev_in and physdev_out parameters

### PR DESCRIPTION
**Pull request by jonnytpuppet**
Added support for iptables physdev_in and physdev_out parameters. This allows puppet to mange firewall rules in bridged environments.

### CODE REVIEW

File: lib/puppet/provider/firewall/iptables.rb [hunk 1]
```diff
@@ -222,7 +224,7 @@ def self.rule_to_hash(line, table, counter)
     ####################
     # PRE-PARSE CLUDGING
     ####################
-
+
```
Comment (hunner) [inline lib/puppet/provider/firewall/iptables.rb]: Trailing spaces.

File: lib/puppet/type/firewall.rb [hunk 2]
```diff
@@ -1191,5 +1205,11 @@ def should_to_s(value)
       self.fail "Parameter 'stat_probability' requires 'stat_mode' to be set to 'random'"
     end
 
+    if value(:bridge)
```
Comment (hunner) [inline lib/puppet/type/firewall.rb]: There is no property called `bridge` yet. Do you want to make one for the ipv6 provider?

File: lib/puppet/type/firewall.rb [hunk 3]
```diff
@@ -1033,6 +1033,20 @@ def should_to_s(value)
     newvalues(/^([0-9a-f]{2}[:]){5}([0-9a-f]{2})$/i)
   end
 
+  newproperty(:physdev_in, :required_features => :iptables) do
+    desc <<-EOS
+      Match if the packet is entering a bridge from the given interface.
+    EOS
+    newvalues(/^[a-zA-Z0-9\-\._\+]+$/)
```
Comment (hunner) [inline lib/puppet/type/firewall.rb]: In ruby regex, `\w` is `[a-zA-Z0-9_]` if you want to use that.

File: lib/puppet/type/firewall.rb [hunk 4]
```diff
@@ -1033,6 +1033,20 @@ def should_to_s(value)
...
+  end
+
+  newproperty(:physdev_out, :required_features => :iptables) do
+    desc <<-EOS
+      Match if the packet is leaving a bridge via the given interface.
```
Comment (hunner) [inline lib/puppet/type/firewall.rb]: These docs are for `--physdev-is-out` not `--physdev-out` as per http://ipset.netfilter.org/iptables-extensions.man.html

Comment (jonnytpuppet) [inline lib/puppet/type/firewall.rb]: --physdev-is-out is a boolean (no interface specified). --physdev-out is tied to a specific, given interface.

File: lib/puppet/provider/firewall/iptables.rb [hunk 5]
```diff
@@ -247,7 +249,7 @@ def self.rule_to_hash(line, table, counter)
         (\s--next)?/x,
         '--pol "ipsec\1\2\3\4\5\6\7\8" '
     )
-
+
```
Comment (hunner) [inline lib/puppet/provider/firewall/iptables.rb]: Trailing spaces.

**Pull request merged**